File: backend/mtg/internal/models/entity/prescription.go
package entity

import (
	"time"

	"github.com/google/uuid"
)

type Prescription struct {
	ID              uuid.UUID        `json:"id" gorm:"type:uuid;primaryKey"`
	Medication      string           `json:"medication"`
	MedicationTypes []MedicationType `json:"medicationType" gorm:"many2many:prescription_medication_types"`
	Dosage          string           `json:"dosage"`
	Notes           *string          `json:"notes"`
	Started         *time.Time       `json:"started" gorm:"type:timestamp;"`
	Ended           *time.Time       `json:"ended" gorm:"type:timestamp;"`
	Refills         *int             `json:"refills"`
	Total           *int             `json:"total" gorm:"type:SMALLINT;default:0"`
	Owner           *string          `json:"owner"`
	PrescribedBy    *uuid.UUID       `json:"prescribedBy"`
	Doctor          *Doctor          `gorm:"foreignKey:PrescribedBy"`
}

File: frontend/src/lib/types/Prescription.ts
import { type MedicationType } from './MedicationType.js';
export type Prescription = {
	id: string;
	medication: string;
	dosage: string;
	notes: string;
	started: string;
	ended: string;
	refills: number;
	total: number;
	prescribedBy: string;
	medicationType: MedicationType[];
};
